### ISSUE TITLE
Items to upstream, and maybe a place to discuss the plan

### ISSUE THREAD

**Issue by klauer** [pcdshub/pcdsdevices#796]
Let's list what we intend to (try to) upstream to ophyd, and who might want to handle the upstream PR.

The ones I know we want to send along:

- `UnitConversionDerivedSignal` (@klauer)
- IPython repr rework (@ZLLentz)
- PVPositioner helpers (@ZLLentz)

Others I'm not sure we've discussed / may take considerable back-and-forth to get accepted:
- `UnrelatedComponent` (@ZLLentz )
- `ObjectComponent`, `InterfaceComponent`, `InterfaceDevice` (@ZLLentz )
- Variety metadata (@klauer)
- `LUTPositioner` (@klauer)
- Tab completion things? (mixed)
- Presets?
- Tweak tools?

Any others missing? Feel free to edit/correct/add to the above, or comment

**Comment by ZLLentz** [comment]
I'd add some more of our utility signals like `InternalSignal` and the metadata edit signals to the second category (perhaps reworked as a mixin?)

I think the further you go down that list the more messy it gets with how much work to make the feature up to par with the rest of the ophyd implementation (looking at you, presets and tweak tools)

**Comment by klauer** [comment]
Added those - and good points.

I guess we can then better prioritize this list based primarily on the two metrics of "does it add a lot of functionality/quality-of-life improvements" and "might it be easy to get merged".

I took a stab at reordering them based on that, though they were mostly there by coincidence.

**Comment by ZLLentz** [comment]
Also I think variety metadata is actually pretty high impact because it influences how people use typhos, and having the machinery that typhos uses exist in ophyd would be a pretty big deal

**Comment by klauer** [comment]
Agreed on variety md impact, but of the above items it probably has the highest chance to sit in a "PR open" state for ages. (*) Perhaps that balances out to being in the middle then? 🤷‍♂️ 

(*) This comment only reflects my opinion; ~no party makes any guarantees or warranties of any kind, express or implied, including, but not limited to, any warranties of merchantability or fitness for a particular purpose, whether arising by operation of law or otherwise.~

**Comment by ZLLentz** [comment]
oh I think I agree entirely on that point

**Comment by klauer** [comment]
Is there any pending work for variety metadata? I can throw up a PR and let it sit there and stew. Perhaps it won't be as bad as I expect *cough*.

**Comment by ZLLentz** [comment]
TBH I cannot remember a thing about it

**Comment by klauer** [comment]
New item: maybe some form of the set current position fixes? (https://github.com/pcdshub/pcdsdevices/pull/806)

**Comment by ZLLentz** [comment]
I think maybe it makes sense to use set_and_wait for all the sets in the parent class, but also the only reason this is an issue is because we're using a non-standard motor record.

**Comment by klauer** [comment]
The small first PR for UnitConversionDerivedSignal took about 2 months to get included in ophyd. We should prioritize this issue appropriately based on that fact.
